crates/ruma-common/tests/events/audio.rs

```diff
@@ -379,3 +381,97 @@ fn message_event_deserialization() {
             && unsigned.is_empty()
     );
 }
+
+#[test]
+fn room_message_serialization() {
+    let message_event_content =
+        RoomMessageEventContent::new(MessageType::Audio(AudioMessageEventContent::plain(
+            "Upload: my_song.mp3".to_owned(),
+            mxc_uri!("mxc://notareal.hs/file").to_owned(),
+            None,
+        )));
+
+    assert_eq!(
+        to_json_value(&message_event_content).unwrap(),
+        json!({
+            "body": "Upload: my_song.mp3",
+            "url": "mxc://notareal.hs/file",
+            "msgtype": "m.audio",
+            "org.matrix.msc1767.text": "Upload: my_song.mp3",
+            "org.matrix.msc1767.file": {
+                "url": "mxc://notareal.hs/file",
+            },
+            "org.matrix.msc1767.audio": {},
+        })
+    );
+}
+
+#[test]
+fn room_message_stable_deserialization() {
+    let json_data = json!({
+        "body": "Upload: my_song.mp3",
+        "url": "mxc://notareal.hs/file",
+        "msgtype": "m.audio",
+        "m.text": "Upload: my_song.mp3",
+        "m.file": {
+            "url": "mxc://notareal.hs/file",
+        },
+        "m.audio": {},
+    });
+
+    assert_matches!(
+        from_json_value::<RoomMessageEventContent>(json_data)
+            .unwrap(),
+        RoomMessageEventContent {
+            msgtype: MessageType::Audio(AudioMessageEventContent {
+                body,
+                src: MediaSource::Plain(url),
+                message: Some(message),
+                file: Some(file),
+                audio: Some(_),
+                ..
+            }),
+            ..
+        } if body == "Upload: my_song.mp3"
+          && url == "mxc://notareal.hs/file"
+          && message.variants().len() == 1
+          && message.variants()[0].body == "Upload: my_song.mp3"
+          && file.url == "mxc://notareal.hs/file"
+          && !file.is_encrypted()
+    );
```

Comment:
Can you write these as smaller asserts in the future? (not in this PR though, unless you really feel like it ^^)

I know I've established and enforced this style before, but I've found a few times to often now that it leads to bad error messages when the test fails plus avoiding `unwrap` in tests is kind of pointless.

Reply:
I agree on the bad error messages. I've had them quite a few times because of copy-paste errors, thankfully the errors were easy to find.

However I'm not sure what that would look like.

Reply:
Something like this: https://github.com/ruma/ruma/commit/8f054ba39c500963d5216dc278f4b3616aebca74

Reply:
And since you can't check every field exhaustively in integration tests anyways, instead of destructuring I'd probably just do

```rust
let msgtype = content.msgtype;
assert_eq!(msgtype.body, "Upload: my_song.mp3");
assert_eq!(msgtype.url, …);
```

Reply:
I did it, but since the msgtype is an enum I still had to match it. I find it a bit ugly to `assert_match` then `if let`, but I think that's what's gonna give us the best error messages.